diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/MagicNumberRule.kt

```diff
@@ -37,11 +38,19 @@ class MagicNumberRule(configRules: List<RulesConfig>) : DiktatRule(
         )
     }
     override fun logic(node: ASTNode) {
-        if (node.elementType == INTEGER_CONSTANT || node.elementType == FLOAT_CONSTANT) {
+        if ((node.elementType == INTEGER_CONSTANT || node.elementType == FLOAT_CONSTANT)
+                && !isTopLevelDeclarationConstant(node)) {
```

Comment:
I believe this should be checked along with `isConstant` parameter; you mix the internal logic of the rule when extracting some of the checks to a higher level